proto/regen/ecocredit/v1alpha1/tx.proto

```diff
@@ -427,3 +452,107 @@ message MsgAllowAskDenom {
 
 // MsgAllowAskDenomResponse is the Msg/AllowAskDenom response type.
 message MsgAllowAskDenomResponse {}
+
+message MsgCreateBasket {
+  // curator is the address of the basket curator who is able to change certain
+  // basket settings.
+  string curator = 1;
+
+  // name will be used to create a bank denom for this basket token of the form
+  // ecocredit:{curator}:{name}.
+  string name = 2;
+
+  // display_name will be used to create a bank Metadata display name for this
+  // basket token of the form ecocredit:{curator}:{display_name}.
+  string display_name = 3;
+
+  // exponent is the exponent that will be used for denom metadata. An exponent
+  // of 6 will mean that 10^6 units of a basket token should be displayed
+  // as one unit in user interfaces.
+  uint32 exponent = 4;
+
+  // admission_criteria is the criteria by which credits can be added to the
+  // basket. Admission criteria will be applied in order and the first criteria
+  // which applies to a credit will determine its multiplier in the basket.
+  repeated AdmissionCriteria admission_criteria = 5;
+
+  bool retire_on_take = 6;
+
+  bool allow_picking = 7;
+}
+
+message AdmissionCriteria {
+  Filter filter = 1;
+
+  // multiplier is an integer number which is applied to credit units when
+  // converting to basket units. For example if the multiplier is 2000, then
+  // 1.1 credits will result in 2200 basket tokens. If there are any fractional
+  // amounts left over in this calculation when adding credits to a basket,
+  // those fractional amounts will not get added to the basket.
+  string multiplier = 2;
```

Comment:
What is this multiplier field used? Is it more appropriate to put it in `MsgCreateBasket` ?

Reply:
```suggestion
  // multiplier is an integer number which is applied to credit units when
  // converting to basket units. For example if the multiplier is 2000, then
  // 1.1 credits will result in 2200 basket tokens. If there are any fractional
  // amounts left over in this calculation when adding credits to a basket,
  // those fractional amounts will not get added to the basket.
  int64 multiplier = 2;
```

should be an int and not string too right?

Reply:
I think I may have understood: credit amounts can be fractional (e.g. `1.1` in the comment), but basket tokens (sdk.Coins) are integers. Multiplier allows to lose less of the decimal parts of ecocredits. Is that correct?

Still think it should be in MsgCreateBasket though

Reply:
> should be an int and not string too right?

Well in cosmos we generally use big integers and those are serialized as strings in protobuf

Reply:
> Ah I think I may have understood: credit amounts can be fractional (e.g. `1.1` in the comment), but basket tokens (sdk.Coins) are integers. Multiplier allows to lose less of the decimal parts of ecocredits. Is that correct?
> 
> Still think it should be in MsgCreateBasket though. We can then delete the comment
> 
> > the first criteria which applies to a credit will determine its multiplier in the basket.

It is only used in `MsgCreateBasket`. This design allows different multipliers for different types of credits within a basket. For instance, someone could say XYZ credit is really 80% as valuable as ABC credit and give them different multipliers within the same basket.

Reply:
renamed `AdmissionCriteria` to `BasketCriteria`. Is that any clearer?